packages/devtools/lib/src/timeline/cpu_profile_protocol.dart

```diff
@@ -15,6 +15,13 @@ class CpuProfileData {
         stackFramesJson = cpuProfileResponse.json['stackFrames'],
         stackTraceEvents = cpuProfileResponse.json['traceEvents'] {
     _processStackFrames(cpuProfileResponse);
+    _setLeafCounts();
+
+    assert(
```

Comment:
What does this assert mean? 
This seems error prone and possibly wrong. Couldn't both counts be zero?

Reply:
It's basically asserting that our logic of counting samples per frame is correct. `sampleCount` is the number we get from the response. Our code should match that.

Yes they could both be 0 if we got a response with no samples. If both counts are zero then the assert shouldn't fail.

Reply:
Oops I see it is `!=` when it should be `==`. It wasn't showing up in the console because this call is nested into a try-catch. I'm going to change this to a print so that if it does fail (which it shouldn't), it fails more gracefully with a message to the console.

Reply:
it is fine to leave as an assert. Dart asserts are ignored in non-debug builds so we won't break users if we have a logic error in an assert but we might catch errors as we build our apps.

Reply:
done.